poethepoet/task/script.py

```diff
@@ -102,10 +103,22 @@ def _resolve_args(cls, call_params: str, named_args: Dict[str, str]):
         def resolve_param(param: str):
             if "=" in param:
                 keyword, value = (token.strip() for token in param.split(r"=", 1))
-                return f"{keyword}={named_args.get(value, value)!r}"
+                return {keyword: named_args.get(value, value)}
             else:
-                return named_args.get(param, param)
-
-        return ", ".join(
-            resolve_param(param.strip()) for param in call_params.strip().split(",")
-        )
+                return {param: named_args.get(param, param)}
+
+        # I only want to filter out the empty strings that can come from the parser of named_args without having to modify named args
+        # test_script_task_omit_kwarg is an example of why this is necessary -- without removing the '' we have inconsistant behavior
+        # even if the arg isn't passed to poe it will be otherwise created here with an '' value
+        return {
+            k: v
+            for k, v in dict(
+                ChainMap(
```

Comment:
I think this could be simpler without ChainMap, am I missing something?
```suggestion
        return {
            key: value
            for param in call_params.split(",")
            for key, value in resolve_param(param.strip()).items()
            if value is not None
        }
```

Also single char variables (especially spanning multiple lines) are a pet peeve of mine.

Reply:
I need to take a look. I can't recall exactly what issue I had when I wasn't using the ChainMap but it was an early bit of code I wrote so I might just not have given it sufficient thought. I'll look into this change.